specification/synapse/data-plane/Microsoft.Synapse/preview/2022-03-01-preview/runNotebook.json

```diff
@@ -0,0 +1,554 @@
+        "sessionDetail": {
+          "description": "Run notebook session details.",
+          "type": "object"
+        },
+        "exitValue": {
+          "description": "Output of exit command.",
+          "type": "string"
+        },
+        "error": {
+          "$ref": "#/definitions/RunNotebookError"
+        }
+      }
+    },
+    "RunNotebookRequest": {
+      "description": "Run notebook request.",
+      "type": "object",
+      "properties": {
+        "notebook": {
+          "type": "string",
+          "description": "Notebook name."
+        },
+        "sparkPool": {
+          "type": "string",
+          "description": "SparkPool name."
+        },
+        "sessionOptions": {
+          "description": "Session properties.",
+          "$ref": "#/definitions/RunNotebookSparkSessionOptions"
+        },
+        "honorSessionTimeToLive": {
+          "description": "Whether session should run till time to live after run completes.",
+          "type": "boolean"
+        },
+        "parameters": {
+          "type": "object",
+          "description": "Run notebook parameters",
+          "additionalProperties": {
+            "$ref": "#/definitions/RunNotebookParameter"
+          }
+        }
+      }
+    },
+    "RunNotebookParameter": {
+      "description": "Run notebook parameter.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Parameter type."
+        },
+        "value": {
+          "type": "object",
+          "description": "Parameter value."
+        }
+      }
+    },
+    "RunNotebookSnapshot": {
+      "description": "Run notebook snapshot.",
+      "type": "object",
+      "properties": {
+        "exitValue": {
+          "description": "Output of exit command.",
+          "type": "string"
+        },
+        "id": {
+          "description": "Run notebook runId.",
+          "type": "string"
+        },
+        "notebook": {
+          "description": "Notebook name.",
+          "type": "string"
+        },
+        "sessionOptions": {
+          "description": "Session properties.",
+          "$ref": "#/definitions/RunNotebookSparkSessionOptions"
+        },
+        "honorSessionTimeToLive": {
+          "description": "Whether session should run till time to live after run completes.",
+          "type": "boolean"
+        },
+        "sessionId": {
+          "description": "Livy session id.",
+          "type": "integer",
+          "format": "int64"
+        },
+        "sparkPool": {
+          "description": "SparkPool name.",
+          "type": "string"
+        },
+        "parameters": {
+          "type": "object",
+          "description": "Run notebook parameters",
+          "additionalProperties": {
+            "$ref": "#/definitions/RunNotebookParameter"
+          }
+        },
+        "notebookContent": {
+          "$ref": "../2021-06-01-preview/notebooks.json#/definitions/NotebookResource"
```

Comment:
Can we refer 2020-12-01 version which the same version in v7 tag?

```
- Microsoft.Synapse/stable/2020-12-01/notebooks.json
```

Reply:
Why do we need to change the version? We have other versions as part of v7 as well.


```
- Microsoft.Synapse/preview/2023-04-18-preview/linkConnections.json
- Microsoft.Synapse/preview/2022-03-01-preview/runNotebook.json
- Microsoft.Synapse/preview/2021-11-01-preview/kqlScripts.json
- Microsoft.Synapse/preview/2021-07-01-preview/symsSync.json
- Microsoft.Synapse/preview/2021-06-01-preview/sparkConfigurations.json
- Microsoft.Synapse/preview/2021-06-01-preview/entityTypes/SparkConfiguration.json
```

Reply:
I notice that we have two NotebookResource defined in our latest v7 tag:

- The first one is referred by runNotebook.json, https://github.com/azure/azure-rest-api-specs/blob/main/specification/synapse/data-plane/Microsoft.Synapse/preview/2022-03-01-preview/runNotebook.json#L348-L350 and the relevant version is /2021-06-01-preview/notebooks.json#/definitions/NotebookResource

- The second one is defined and used in notebooks.json, https://github.com/Azure/azure-rest-api-specs/blob/main/specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/notebooks.json#L356-L382

These two references would cause two duplicated schema for the same model name `NotebookResource` which finally introduces one to be renamed. For JS this may be a potential breaking.

Reply:
Thank you @MaryGao, sent a PR https://github.com/Azure/azure-rest-api-specs/pull/24493 to fix the potential breaking, please help review, thanks.